R/set_synth_years.R

```diff
@@ -1,58 +1,61 @@
 
+  spawn_years <- DSMCalibrationData::calibration_year_spawn_index
   years <- DSMCalibrationData::calibration_year_index
 
-  inputs$freeport_flows <- inputs$freeport_flows[ , years]
-  inputs$vernalis_flows <- inputs$vernalis_flows[ , years]
-  inputs$stockton_flows <- inputs$stockton_flows[ , years]
-  inputs$CVP_exports <- inputs$CVP_exports[ , years]
-  inputs$SWP_exports <- inputs$SWP_exports[ , years]
-  inputs$proportion_diverted <- inputs$proportion_diverted[ , , years]
-  inputs$total_diverted <- inputs$proportion_diverted[ , , years]
-  inputs$delta_proportion_diverted <- inputs$delta_proportion_diverted[ , years, ]
-  inputs$delta_total_diverted <- inputs$delta_total_diverted[ , years, ]
-  inputs$prop_flow_natal <- inputs$prop_flow_natal[ , years] # extra year
-  inputs$upper_sacramento_flows <- inputs$upper_sacramento_flows[ , years]
-  inputs$delta_inflow <- inputs$delta_inflow[ , years, ]
-  inputs$proportion_flow_bypass <- inputs$proportion_flow_bypass[ , years, ]
-  inputs$gates_overtopped <- inputs$gates_overtopped[ , years, ]
-  inputs$vernalis_temps <- inputs$vernalis_temps[ , years] # extra year
-  inputs$prisoners_point_temps <- inputs$prisoners_point_temps[ , years] # extra year
-  inputs$degree_days <- inputs$degree_days[ , , years]
-  inputs$avg_temp <- inputs$avg_temp[ , , years] # extra year
-  inputs$avg_temp_delta <- inputs$avg_temp_delta[ , years, ] # extra year
-  inputs$spawning_habitat <- inputs$spawning_habitat[ , , years] # extra year
-  inputs$inchannel_habitat_fry <- inputs$inchannel_habitat_fry[ , , years]
-  inputs$inchannel_habitat_juvenile <- inputs$inchannel_habitat_juvenile[ , , years]
-  inputs$floodplain_habitat <- inputs$floodplain_habitat[ , , years]
-  inputs$weeks_flooded <- inputs$weeks_flooded[ , , years]
-  inputs$delta_habitat <- inputs$delta_habitat[ , years, ]
-  inputs$sutter_habitat <- inputs$sutter_habitat[ , years]
-  inputs$yolo_habitat <- inputs$yolo_habitat[ , years]
+  params$freeport_flows <- params$freeport_flows[ , years]
+  params$vernalis_flows <- params$vernalis_flows[ , years]
+  params$stockton_flows <- params$stockton_flows[ , years]
+  params$CVP_exports <- params$CVP_exports[ , years]
+  params$SWP_exports <- params$SWP_exports[ , years]
+  params$proportion_diverted <- params$proportion_diverted[ , , years]
+  params$total_diverted <- params$proportion_diverted[ , , years]
+  params$delta_proportion_diverted <- params$delta_proportion_diverted[ , years, ]
+  params$delta_total_diverted <- params$delta_total_diverted[ , years, ]
+  params$prop_flow_natal <- params$prop_flow_natal[ , spawn_years]
+  params$upper_sacramento_flows <- params$upper_sacramento_flows[ , years]
+  params$delta_inflow <- params$delta_inflow[ , years, ]
+  params$proportion_flow_bypass <- params$proportion_flow_bypass[ , years, ]
+  params$gates_overtopped <- params$gates_overtopped[ , years, ]
+  params$vernalis_temps <- params$vernalis_temps[ , years] # should this have spawn_years
```

Comment:
@ergz and @ErinCain for the few inputs that I have the comment `# should this have spawn_years`, can you plz confirm what years it should be. The originals start 1979-2000, but should they be 1980-2000? Or does the indexing need to be adjusted in the model where they are used? For proxy should they be 1997-2017 or 1998-2017?

Reply:
Looks like `vernalis_temps`, `prisoners_point_temps`, `avg_temp`, and  `avg_temp_delta` are only used in rearing survival and migratory survival functions (these dynamics are all months 1-8) so we should not need the `spawn_years` for these

Reply:
thanks for updating the temperature package to remove 1979 data where it was not applicable I removed these comments.